Mathlib/Computability/ContextFreeGrammar.lean

```diff
@@ -253,3 +253,176 @@ theorem Language.IsContextFree.reverse (L : Language T) :
   fun ⟨g, hg⟩ => ⟨g.reverse, hg ▸ Set.ext g.mem_reverse_language_iff_reverse_mem_language⟩
 
 end closure_reversal
+
+section embed_project
+
+/-! This section contains only auxiliary constructions that will shorten upcoming proofs of
+closure properties. When combining several grammars together, we usually want to take a sum type of
+their nonterminal types and embed respective nonterminals to this sum type. We subsequently show
+that the resulting grammar preserves derivations of those strings that may contain any terminals but
+only the proper nonterminals. The embedding operation must be injective. The projection operation
```

Comment:
Using "terminal" and "nonterminal" in place of "terminal symbol" and "nonterminal symbol" is quite idiosyncratic and definitely hampers comprehension
```suggestion
that the resulting grammar preserves derivations of those strings that may contain any terminal symbols but
only the proper nonterminal symbols. The embedding operation must be injective. The projection operation
```
and below

Reply:
Are you sure that you want to use "terminal symbols" and "nonterminal symbols" only?

In the ITP paper that documents the Lean 3 library, I was naming it as follows:
let `(n : g.NT)`
`n` .. nonterminal
`Symbol.nonterminal n` ... nonterminal symbol